source/endpoints/devices/Devices.py

```diff
@@ -15,22 +15,19 @@
 class Devices(Resource):
     """ Shows a list of all devices and adds a new device with POST """
 
+
     @namespace.doc("list_devices")
     @namespace.marshal_list_with(device)
     def get(self):
         """ List all devices """
-        return device_database.get_devices()
+        return business_logic_devices.get_all_devices_from_database()
 
     @namespace.doc("post_device")
     @namespace.expect(format_post_device)
     @namespace.response(201, "new device")
     def post(self):
         """ Post a new device """
-        required_info = namespace.apis[0].payload["required_info"]
-        new_device_info = eval(required_info)
-        name = new_device_info.pop("name", None)
-        class_plugin = plugin_manager.get_implementation_of(new_device_info)
-        class_plugin.name = name
-        device_database.add_device(class_plugin)
-        class_plugin.setup()
+        json_required_info = namespace.apis[0].payload["required_info"]
+        plugin = business_logic_devices.create_new_device_from_json_input(json_required_info)
```

Comment:
Now there is still some business logic in the endpoint with two calls to the business logic class,